### PR TITLE
run `eslint .` plus .eslintignore

### LINKED ISSUE #7770: make lint:js run `eslint .` plus .eslintignore
https://github.com/ember-cli/ember-cli/pull/7769#issuecomment-382450832

```js
    "lint:js": "eslint ./*.js app blueprints config lib server tests",
```

```js
    "lint:js": "eslint .",
```

cc @rwjblue @dfreeman

### PR DESCRIPTION
**Pull request by kellyselden**
Closes #7770

node_modules appears to be auto ignored, so no override necessary.

### CODE REVIEW

File: tests/unit/blueprints/addon-test.js [hunk 1]
```diff
@@ -70,7 +70,6 @@ describe('blueprint - addon', function() {
     "ember-addon"\n\
   ],\n\
   "scripts": {\n\
-    "lint:js": "eslint ./*.js addon addon-test-support app blueprints config lib server test-support tests",\n\
```
Comment (Turbo87) [inline tests/unit/blueprints/addon-test.js]: this seems wrong to me. the line should still exist, but look a little different 🤔

File: blueprints/app/files/.eslintignore [hunk 2]
```diff
@@ -1 +1,3 @@
 /blueprints/*/files/**/*.js
+/dist/
+/tmp/
```
Comment (Turbo87) [inline blueprints/app/files/.eslintignore]: related to https://github.com/ember-cli/ember-cli/pull/7663 we might want to add `vendor` here too

File: tests/unit/blueprints/addon-test.js (hunk 1 above)
Comment (rwjblue) [reply tests/unit/blueprints/addon-test.js]: No, I think it is correct. This test is only testing the changes programmatically made by the addon blueprint. If you take a look at the fixtures above (addon, app, MU app, etc) you will see that we still have the `lint:js` script present...

File: blueprints/app/files/.eslintignore (hunk 2 above)
Comment (rwjblue) [reply blueprints/app/files/.eslintignore]: Agreed.

File: tests/unit/blueprints/addon-test.js (hunk 1 above)
Comment (Turbo87) [reply tests/unit/blueprints/addon-test.js]: oh, it's the diff... nevermind 🙈

**Pull request merged**